wgpu-core/src/lock/ranked.rs

```diff
@@ -281,6 +281,15 @@ impl<T> RwLock<T> {
     }
 }
 
+impl<'a, T> RwLockWriteGuard<'a, T> {
+    pub fn downgrade(this: Self) -> RwLockReadGuard<'a, T> {
+        RwLockReadGuard {
```

Comment:
nitpick(non-blocking): Destructuring might let us catch fields we add later, if ever.

<sub>I'm gonna go write some feedback that maybe actually is important now. 👉🏻👈🏻</sub>